### DISCUSSION TITLE
Choosing Between Version 1 and Version 2

### DISCUSSION THREAD

**Discussion by EzizAnnagurbanow** [flyerhq/flutter_chat_ui#694] [Q&A] [answered]
Hi,

I'm building a feature-rich messenger and need to decide between version 1 and version 2 of this library. Version 1 seems more stable, but version 2 offers advanced features and better customization. For a production-level app, would you recommend prioritizing stability or focusing on the newer features of version 2?

Thank you!

**Comment by demchenkoalex** [comment, marked as answer]
Hello! 👋

I do not recommend using `v1`, as it has been unsupported for a long time. You will eventually need to migrate to `v2`, which will not have a migration guide because it was written from scratch. In fact, these two versions take entirely different approaches.

If you’re looking for a feature-rich messenger, `v2` is a much better choice. It’s designed to eventually support all standard messenger functions, is significantly more performant, and offers more extensive and straightforward customisation options. However, please note that the documentation is not ready yet, and I’m still working on implementing features from `v1` (as tracked in a pinned issue). This means `v2` might currently lack some important functionality.

If you’re comfortable exploring the source code and examples to understand how things work, you should be fine using `v2` as a base to build upon.

Otherwise, I’d recommend forking or considering alternative packages if the current state of `v2` doesn’t meet your needs.

**Comment by EzizAnnagurbanow** [comment]
Thank you for reply, i think really good idea using v2
